src/workloads/query/ShardFilter.yml

```diff
@@ -0,0 +1,101 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests the performance of queries which need to perform shard filtering.
+Keywords:
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  Namespace: &Namespace test.Collection0
+  MaxPhases: &MaxPhases 8
+  DocumentCount: &DocumentCount 1000
+
+Actors:
+- Name: CreateShardedCollection
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: admin
+        Operations:
+        - OperationMetricsName: EnableSharding
+          OperationName: AdminCommand
+          OperationCommand:
+            enableSharding: *Database
+        - OperationMetricsName: ShardCollection
+          OperationName: AdminCommand
+          OperationCommand:
+            shardCollection: *Namespace
+            key: {sKey: 1}
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        Collection: *Collection
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: *DocumentCount
+        Document:
+          sKey: {^RandomInt: {min: 0, max: 10000}}
+          a: {^RandomInt: {min: 0, max: 10000}}
+
+- Name: QueryByShardKey
+  Type: CrudActor
+  Database: *Database
+  Threads: 16
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 1 minute
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {sKey: {$gt: 5000}}
+
+- Name: QueryByNonShardKey
+  Type: CrudActor
+  Database: *Database
+  Threads: 16
+  Phases:
+    OnlyActiveInPhases:
+      Active: [3]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 1 minute
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {a: {$gt: 5000}}
+
+- Name: QueryByShardKeyExpression
+  Type: CrudActor
+  Database: *Database
+  Threads: 16
+  Phases:
+    OnlyActiveInPhases:
+      Active: [4]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: 1 minute
+        Collection: *Collection
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter: {sKey: {$in: [123, 5000, 6789]}}
```

Comment:
Shouldn't we have an AutoRun section here? This should include the bonsai 3-shard variant, as well as the baseline 3-shard variant if we want it to run in waterfall with Bonsai + master so we can compare the two in the dashboard. ([example](https://github.com/mongodb/genny/blob/9af1ccd6887c5320cd08801b6bc02a1b9a13af83/src/workloads/sharding/MultiShardTransactionsWithManyNamespaces.yml#L243-L248))

That's probably why it didn't run in the patch you linked.

Reply:
Perfect, thanks! Small request to run the 3-shard version of your workload in your patch as well, as a sanity check.

Reply:
https://spruce.mongodb.com/version/651dbed03627e0894809bf3b/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC